### PR TITLE
Feature/delete spending query

### PR DESCRIPTION
**Pull request by anastasiya-nd**

### CODE REVIEW

File: src/api/api.js [hunk 1]
```diff
@@ -13,6 +13,13 @@ const spendings = {
       },
     });
   },
+  deleteSpending(id) {
+    return instance.delete(`/spendings/${id}`, {
+      params: {
```
Comment (ok2ju) [inline src/api/api.js:18]: Why do you need `id` in the params? You already have it in the url

File: src/components/SpendingItem/SpendingItem.jsx [hunk 2]
```diff
@@ -49,13 +49,15 @@ const SpendingItem = ({ category, note, labels, createdAt, amount, currency }) =
...
     }
   };
-
   const getFormatDate = (date) => {
     const formatDate = new Date(date);
     return `${
       shortMonthNames[formatDate.getMonth()]
     } ${formatDate.getDate()}, ${formatDate.getFullYear()}`;
   };
+  const deleteSpending = (spendingID) => {
```
Comment (ok2ju) [inline src/components/SpendingItem/SpendingItem.jsx:58]: In most cases handler for `on...` events can be called `handle...`. E.g. `handleDeleteSpending`

File: src/redux/reducer.js [hunk 3]
```diff
@@ -33,6 +36,19 @@ const reducer = (state = initialState, action) => {
         ...state,
         loading: false,
       };
+    case DELETE_SPENDING_PENDING:
+      return {
+        ...state,
```
Comment (ok2ju) [inline src/redux/reducer.js:41]: Toggle `loading` to `true` here.

File: src/redux/reducer.js [hunk 4]
```diff
@@ -33,6 +36,19 @@ const reducer = (state = initialState, action) => {
...
+      };
+    case DELETE_SPENDING_SUCCESS:
+      return {
+        ...state,
+        spendings: state.spendings.filter((spending) => spending._id !== action.payload.id), // eslint-disable-line
+      };
```
Comment (ok2ju) [inline src/redux/reducer.js:47]: Don't forget to set `loading: false` here

File: src/redux/reducer.js [hunk 5]
```diff
@@ -33,6 +36,19 @@ const reducer = (state = initialState, action) => {
...
+    case DELETE_SPENDING_ERROR:
+      return {
+        ...state,
```
Comment (ok2ju) [inline src/redux/reducer.js:50]: Set `loading: false` here.

File: src/redux/sagas.js [hunk 6]
```diff
@@ -13,6 +18,18 @@ function* getSpendingsWorker(data) {
   }
 }
 
+function* deleteSpendingWorker(data) {
```
Comment (ok2ju) [inline src/redux/sagas.js:21]: Please, rename `data` to `action`

File: src/redux/sagas.js [hunk 7]
```diff
@@ -13,6 +18,18 @@ function* getSpendingsWorker(data) {
...
+  try {
+    yield call(spendings.deleteSpending, data.payload.id);
+    yield put(deleteSpendingSuccess(data.payload.id));
+  } catch {
```
Comment (ok2ju) [inline src/redux/sagas.js:25]: `catch (error)`. Also pass `error.message` to `deleteSpendingError` action-creator

File: package.json [hunk 8]
```diff
@@ -26,6 +26,7 @@
     "@babel/preset-env": "^7.15.0",
     "@babel/preset-react": "^7.14.5",
     "babel-loader": "^8.2.2",
+    "babel-plugin-styled-components": "^1.13.2",
```
Comment (ok2ju) [inline package.json:29]: Is it works now?

**Review by ok2ju — CHANGES_REQUESTED** [review]

File: src/redux/sagas.js (hunk 7 above)
Comment (anastasiya-nd) [reply src/redux/sagas.js:25]: > `catch (error)`. Also pass `error.message` to `deleteSpendingError` action-creator

I have done it, but I'm not sure that it's right

File: src/components/SpendingItem/SpendingItem.jsx [hunk 9]
```diff
@@ -27,6 +27,11 @@ const SpendingItem = ({ category, note, labels, createdAt, amount, currency }) =
     'Nov',
     'Dec',
   ];
+
+  const formatCategory = (categoryName) => {
+    return categoryName[0].toUpperCase() + categoryName.slice(1);
```
Comment (ok2ju) [inline src/components/SpendingItem/SpendingItem.jsx:32]: I don't like this fn. It's better to create a mapper here, e.g.:
```js
const categoryLabelMapper = {
  food: 'Food',
  ...: '...',
};

<Styled.CategoryName>{categoryLabelMapper[category]}</Styled.CategoryName>
```

File: src/pages/Home/Home.jsx [hunk 10]
```diff
@@ -9,48 +9,84 @@ import Modal from '../../components/Modal';
...
     toggleModal(true);
   };
-
   const handleCloseModal = () => {
     toggleModal(false);
   };
 
+  const handleOpenDeletingModal = () => {
+    toggleDeletingModal(true);
+  };
```
Comment (ok2ju) [inline src/pages/Home/Home.jsx:31]: Missing newline here

File: src/pages/Home/Home.jsx [hunk 11]
```diff
@@ -9,48 +9,84 @@ import Modal from '../../components/Modal';
...
+  const handleCloseDeletingModal = () => {
+    toggleDeletingModal(false);
+  };
+
+  const onDelete = (id) => {
+    setSpendingID(id);
+    handleOpenDeletingModal();
+  };
```
Comment (ok2ju) [inline src/pages/Home/Home.jsx:39]: Missing newline here

**Pull request merged**